packages/@ember/-internals/routing/lib/system/route.ts

```diff
@@ -1874,7 +1874,7 @@ class Route<T = unknown>
     let controllerName = this.controllerName || this.routeName;
     let owner = getOwner(this);
     assert('Route is unexpectedly missing an owner', owner);
-    let controller = owner.lookup<Controller>(`controller:${controllerName}`);
+    let controller = owner.lookup(`controller:${controllerName}`) as Controller;
```

Comment:
**~:** One thing this flags up, just looking at this whole section, is how important dev assertions actually are here. We have, AFAICT, *zero* guarantee from the types here that the `controllerName` is actually a controller. We have *some* guarantees from tests of the runtime, but there's nothing which the compiler can check.